dfget/core/uploader/peer_server_executor.go

```diff
@@ -70,6 +72,12 @@ type peerServerExecutor struct {
 var _ PeerServerExecutor = &peerServerExecutor{}
 
 func (pe *peerServerExecutor) StartPeerServerProcess(cfg *config.Config) (port int, err error) {
+	fileLock := fileutils.NewFileLock(filepath.Dir(cfg.RV.MetaPath))
+	if err = fileLock.Lock(); err != nil {
```

Comment:
It's better to run ` fileLock.Lock()` after the `pe. checkPeerServerExist()` below.